frontend/src/components/elements/Table/Table.tsx

```diff
@@ -16,131 +16,95 @@
  */
 
 import React, { ReactElement } from "react"
-import { Table as ReactTable } from "reactstrap"
 import { toFormattedString } from "lib/format"
 import { Map as ImmutableMap } from "immutable"
 import { dataFrameGet, dataFrameGetDimensions } from "lib/dataFrameProto"
 import withFullScreenWrapper from "hocs/withFullScreenWrapper"
 import "./Table.scss"
+import { range } from "lodash"
+
+type DataFrame = ImmutableMap<string, any>
 
 /**
  * Functional element representing a DataFrame.
  */
 export interface TableProps {
-  element: ImmutableMap<string, any>
+  element: DataFrame
+}
+
+function generateTableCell(
+  df: DataFrame,
+  rowIdx: number,
+  colIdx: number
+): ReactElement {
+  const { contents, styles, type } = dataFrameGet(df, colIdx, rowIdx)
+  const formattedContents = toFormattedString(contents)
+
+  switch (type) {
+    case "corner":
+      return <td key={colIdx}>&nbsp;</td>
+    case "row-header":
+      return (
+        <th key={colIdx} scope="row">
+          {formattedContents}
+        </th>
+      )
+    case "col-header":
+      return (
+        <th scope="column" key={colIdx}>
+          {formattedContents}
+        </th>
+      )
+    case "data":
+      return (
+        <td key={colIdx} style={styles}>
+          {formattedContents}
+        </td>
+      )
+    default:
+      throw new Error(`Cannot parse type "${type}".`)
+  }
+}
+
+function generateTableRow(
+  df: DataFrame,
+  rowIdx: number,
+  cols: number
+): ReactElement {
+  return (
+    <tr key={rowIdx}>
+      {range(cols).map(colIdx => generateTableCell(df, rowIdx, colIdx))}
+    </tr>
+  )
 }
 
 export function Table({ element }: TableProps): ReactElement {
   const { headerRows, rows, cols } = dataFrameGetDimensions(element)
+  const allRows = range(rows)
+  const columns = allRows.slice(0, headerRows)
```

Comment:
The naming just confused me for a bit. I was thinking it'd be column data but it looks like it's basically the indices of number of headerRows.

Reply:
I guess I can call it headerRows, but it's taken lol, maybe `columnHeaders`